### PR TITLE
feature: Add a Sample App for AvaloniaUI

### PR DESCRIPTION
**Pull request by worldbeater**
**What kind of change does this PR introduce?**

This PR adds a sample app and an `IView` implementation for Avalonia. Related issue: https://github.com/reactiveui/Sextant/issues/252

**What is the current behavior?**

Currently, there is now Avalonia support.

**What is the new behavior?**

Now, there is a sample app demonstrating how to use Avalonia and Sextant.

**What might this PR break?**

Nothing.

**Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

Haven't yet added a new `Sextant.Avalonia` project, so here is just a sample app with `IView` implementation. The `NavigationView` for Avalonia implementation still needs tweaking e.g. we need to show background for a model navigation layout when its navigation stack isn't empty.

### CODE REVIEW

File: Sample/SextantSample.Avalonia/App.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            this IMutableDependencyResolver dependencyResolver, Func<TView> navigationViewFactory)
+            where TView : IView
+        {
+            var navigationView = navigationViewFactory();
+            var viewStackService = new ViewStackService(navigationView);
+
+            dependencyResolver.RegisterLazySingleton<IViewStackService>(() => viewStackService);
+            dependencyResolver.RegisterLazySingleton<IView>(() => navigationView, "NavigationView");
+            return dependencyResolver;
+        }
```
Comment (worldbeater) [inline Sample/SextantSample.Avalonia/App.xaml.cs:57]: Copy-pasted this extension method from the `Sextant.XamForms` package and it just worked. Probably we should consider moving this particular extension method to `Sextant` core project as this method seems to be platform-agnostic.

File: Sample/SextantSample.Avalonia/App.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            this IReadonlyDependencyResolver dependencyResolver,
+            string contract = null)
+        {
+            if (dependencyResolver is null)
+            {
+                throw new ArgumentNullException(nameof(dependencyResolver));
+            }
+
+            return dependencyResolver.GetService<IView>(contract ?? "NavigationView");
+        }
```
Comment (worldbeater) [inline Sample/SextantSample.Avalonia/App.xaml.cs:69]: Also copy-pasted this extension method from `Sextant.XamForms` and changed the return type from `NavigationView` to `IView`. And it also just worked. Probably we should consider this extension method to our platform-agnostic project also.

**Comment by RLittlesII** [comment]
@worldbeater Thanks for the contribution, sir!  I am going to get CI figured out before I merge this PR.  We can use this PR as a way to build a full `Sextant.Avalonia` project!

File: Sample/SextantSample.Avalonia/Views/GreenView.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        public GreenView()
+        {
+            AvaloniaXamlLoader.Load(this);
+            this.WhenActivated(disposables =>
+            {
+                this.WhenAnyValue(x => x.ViewModel)
+                    .BindTo(this, x => x.DataContext)
+                    .DisposeWith(disposables);
+            });
```
Comment (worldbeater) [inline Sample/SextantSample.Avalonia/Views/GreenView.xaml.cs:19]: This weird code in `ReactiveUserControl`s is required to keep the `ViewModel` property in sync with `DataContext`. Currently, AvaloniaUI [listens to `DataContextChanged` and updates the view model](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.ReactiveUI/ReactiveUserControl.cs#L23), but not vice versa. Going to think about how to better resolve this on the Avalonia side.

**Comment by worldbeater** [comment]
@RLittlesII thank you for doing this! Going to fix the weird issue related to `ViewModel` and `DataContext` synchronization on the Avalonia side in order to avoid that boilerplate code in view classes derived from `ReactiveUserControl<TViewModel>`. This should make the use of the future `Sextant.Avalonia` project even easier.

File: Sample/SextantSample.Avalonia/Views/GreenView.xaml.cs (hunk 3 above)
Comment (worldbeater) [reply Sample/SextantSample.Avalonia/Views/GreenView.xaml.cs:19]: https://github.com/AvaloniaUI/Avalonia/pull/4917 is now merged, so hopefully, we'll be able to use a preview build soon and refactor this.

**Pull request merged**